diff --git a/libweston/backend-rdp/rdp.c b/libweston/backend-rdp/rdp.c
index 84926745..e5a515f9 100644
--- a/libweston/backend-rdp/rdp.c
+++ b/libweston/backend-rdp/rdp.c
@@ -1634,7 +1634,9 @@ rdp_peer_init(freerdp_peer *client, struct rdp_backend *b)
 	} else {
 		settings->TlsSecurity = FALSE;
 	}
-	settings->NlaSecurity = FALSE;
+	settings->Authentication = TRUE;
+	settings->NlaSecurity = TRUE;
+	settings->RdpSecurity = TRUE;
 
 	if (!client->Initialize(client)) {
 		weston_log("peer initialization failed\n");
